bitacross-worker/service/src/main_impl.rs

```diff
@@ -195,6 +196,11 @@ pub(crate) fn main() {
 		setup::generate_signing_key_file(enclave.as_ref());
 		let tee_accountid = enclave_account(enclave.as_ref());
 		println!("Enclave signing account: {:}", &tee_accountid.to_ss58check());
+	} else if matches.is_present("scheduled-enclave") {
+		let mrenclave = enclave.get_fingerprint().unwrap();
+		println!("check mrenclave, {:?}", &mrenclave.0);
+		&GLOBAL_SCHEDULED_ENCLAVE.init(mrenclave.0).unwrap();
```

Comment:
it seems like getting result via RPC works fine, but if I want to obtain data via binary it returns empty result and sometimes fails to run. Do you know why it might happen? @kziemianek @felixfaisal 

on the screenshot it shows when triggering binary several times, some executions are failing and at the end I've got data via RPC and it works fine. Might it mean that `GLOBAL_SCHEDULED_ENCLAVE.registry` is every time different?


<img width="797" alt="image" src="https://github.com/litentry/litentry-parachain/assets/2113554/909981c3-2850-4d25-a7fb-4d80315715e0">

Reply:
The error implies a race condition.

`Component factory has not been set` means you try to access the global variable before it's initialized (most likely GLOBAL_SCHEDULED_ENCLAVE in this case)

Reply:
It's not necessarily your code's problem - it can be "you simply run the test binary too early"

Reply:
In general I recommend adding more lifecycle events representating the worker state - currently the only thing we have is `EnclaveAdded`

Reply:
It seems that this issue `Component factory has not been set` happens with other commands as well. Do we need need to create an issue for that or not worth it? as it seems really low impact